frappe/public/js/frappe/form/controls/link.js

```diff
@@ -184,7 +184,7 @@ frappe.ui.form.ControlLink = class ControlLink extends frappe.ui.form.ControlDat
 		this.$input.cache = {};
 
 		this.awesomplete = new Awesomplete(me.input, {
-			tabSelect: true,
+			tabSelect: false, // @a11y: Ensure tab navigation works in forms
```

Comment:
Umm, this was deliberately added to improve keyboard navigation 

https://github.com/frappe/frappe/pull/23319

Reply:
Ah yes I understand you're right!

> If a combo box has focus and I type the letter 'T' and see 'Texas' highlighted then I should be able to hit ENTER or TAB at this point and get the same result. I don't think you are going to find anything official on this but I can find bug reports when this isn't the case. ⋅ https://ux.stackexchange.com/a/68635

---

**However**, in a modal window, **there is no way to leave a Link field empty** (I have 500ms before pressing <kbd>Tab</kbd> again) or if I'm navigating to the field a second time (cached results), because pressing <kbd>Esc</kbd> will close the modal instead of just closing the awesomplete.

This means that, in a modal window, I have 500ms to listen (if I'm blind) to the whole field label + description before being **permanently unable to leave a Link field empty**.

https://github.com/frappe/frappe/assets/10946971/1792f0e9-7092-43d8-b3d0-7dde44e9e72a

Also, I'm working on top of a fork (#22689) where the awesomplete opens up **~immediately**, so this means that <kbd>Tab</kbd> will always **_select_** the first item when I just want to _navigate_ in the form. To select I would use <kbd>Enter</kbd> then press <kbd>Tab</kbd> to go to the next field.

Reply:
Interesting aspect. In this case, I think a single `Esc` should only close the opened select, not the whole dialog. Only a second `Esc` should then close the dialog.

Reply:
Thought about it again.
Among devops, enabling `tabSelect` was a widely celebrated UI improvement, so reverting it will be quite a bit disappointing for some. But for end users this may vary.
I guess we should add an “Accessibility” section to System Settings, with the less accessible solution “Tab to select first list item.” turned off or a more accessible solution “Require Enter to select list item.” turned on.
Either way, `Esc` should only close the opened select.

Reply:
@cogk is it only a problem in modals? If so maybe we can _somehow_ fix it instead of disabling it completely.

Reply:
moved to #23391